pkg/kubectl/resource/builder.go

```diff
@@ -120,29 +118,30 @@ func (b *Builder) URL(urls ...*url.URL) *Builder {
 	return b
 }
 
-// Stdin will read objects from the standard input. If ContinueOnError() is set
-// prior to this method being called, objects in the stream that are unrecognized
-// will be ignored (but logged at V(2)).
-func (b *Builder) Stdin() *Builder {
-	return b.Stream(os.Stdin, "STDIN")
-}
-
 // Stream will read objects from the provided reader, and if an error occurs will
 // include the name string in the error message. If ContinueOnError() is set
 // prior to this method being called, objects in the stream that are unrecognized
 // will be ignored (but logged at V(2)).
 func (b *Builder) Stream(r io.Reader, name string) *Builder {
 	b.stream = true
-	b.paths = append(b.paths, NewStreamVisitor(r, b.mapper, b.schema, name, b.continueOnError))
+	b.paths = append(b.paths, &StreamVisitor{r, b.mapper, name, b.continueOnError, b.schema})
 	return b
 }
 
-// Path is a set of filesystem paths that may be files containing one or more
-// resources. If ContinueOnError() is set prior to this method being called,
-// objects on the path that are unrecognized will be ignored (but logged at V(2)).
+// Path accepts a set of paths that may be files, directories or stdin (all can containing
+// one or more resources). Creates a FileVisitor for each file (or stdin) and then each
+// FileVisitor is streaming the content to a StreamVisitor. If ContinueOnError() is set
+// prior to this method being called, objects on the path that are unrecognized will be
+// ignored (but logged at V(2)).
 func (b *Builder) Path(paths ...string) *Builder {
 	for _, p := range paths {
-		i, err := os.Stat(p)
+		if p == "-" {
```

Comment:
This check was in FilenameParam before in order to ensure that someone who wanted to call builder on a path "-" could do so, while calling FilenameParam with "-" means STDIN.  I think we want to preserve that behavior, so the logic for this still needs to be in FilenameParam